libfwupdplugin/fu-common.c

```diff
@@ -159,11 +159,47 @@ fu_common_check_full_disk_encryption(GError **error)
 				    g_variant_get_bytestring(device));
 			return FALSE;
 		}
-		/* TODO identify Ubuntu Core FDE volumes */
 	}
 	return TRUE;
 }
 
+/**
+ * fu_common_check_ubuntu_style_full_disk_encryption:
+ * @error: (nullable): return location for an error
+ *
+ * Checks for the existence of an Ubuntu-style FDE setup, which indicates that we should attempt to
+ * coordinate with snapd to ensure that the FDE volumes are not going to be affected by a firmware
+ * update.
+ *
+ * Returns: %TRUE if an Ubuntu-style FDE setup is detected, or %FALSE otherwise
+ *
+ * Since: 2.0.4
+ **/
+gboolean
+fu_common_check_ubuntu_style_full_disk_encryption(GError **error)
```

Comment:
having ubuntu specific call may be a bit awkward here. Would it be possible to merge this code with fu_common_check_full_disk_encryption()? Maybe with a callback to process devices one by one, so that we can detect presence of either kind of FDE without having 2 marginally different copies of the same code.